jans-cedarling/cedarling/src/jwt/mod.rs

```diff
@@ -368,6 +371,22 @@ impl JwtService {
             .map(|config| &config.policy)
             .cloned()
     }
+
+    /// Find the token metadata key for a given entity type name
+    /// e.g., "Dolphin::Access_Token" -> "access_token"
+    fn find_token_metadata_key(&self, entity_type_name: &str) -> String {
+        // Look through all trusted issuers to find the matching entity type name
+        for (_, issuer_config) in &self.issuer_configs {
+            for (token_key, token_metadata) in &issuer_config.policy.token_metadata {
+                if token_metadata.entity_type_name == entity_type_name {
+                    return token_key.clone();
+                }
+            }
+        }
+        
+        // If not found, return the original mapping (fallback)
+        entity_type_name.to_string()
+    }
```

Comment:
Actually we can return `&str` here:
```suggestion
    fn find_token_metadata_key<'a>(&'a self, entity_type_name: &'a str) -> &'a str {
        // Look through all trusted issuers to find the matching entity type name
        for (_, issuer_config) in &self.issuer_configs {
            for (token_key, token_metadata) in &issuer_config.policy.token_metadata {
                if token_metadata.entity_type_name == entity_type_name {
                    return token_key;
                }
            }
        }

        // If not found, return the original mapping (fallback)
        entity_type_name
    }
```